docs/running-on-yarn.md

```diff
@@ -196,6 +196,15 @@ Most of the configs are the same for Spark on YARN as for other deployment modes
   It should be no larger than the global number of max attempts in the YARN configuration.
   </td>
 </tr>
+<tr>
+  <td><code>spark.yarn.submit.waitAppCompletion</code></td>
```

Comment:
My opinion is that it would be fairly reasonable for users to have relied on the current behavior, so we shouldn't break it.